model/lifecycle.go

```diff
@@ -1426,23 +1421,31 @@ func AddNewTasks(ctx context.Context, activated bool, v *Version, p *Project, pa
 	taskIds := []string{}
 	for _, b := range builds {
 		// Find the set of task names that already exist for the given build
-		tasksInBuild, err := task.Find(task.ByBuildId(b.Id).WithFields(task.DisplayNameKey))
+		tasksInBuild, err := task.Find(task.ByBuildId(b.Id))
 		if err != nil {
 			return nil, err
 		}
 		// build an index to keep track of which tasks already exist
-		existingTasksIndex := map[string]bool{}
+		existingTasksIndex := map[string]task.Task{}
 		for _, t := range tasksInBuild {
-			existingTasksIndex[t.DisplayName] = true
+			existingTasksIndex[t.DisplayName] = t
 		}
 		// if the patch is activated, treat the build as activated
 		b.Activated = activated
 
 		// build a list of tasks that haven't been created yet for the given variant, but have
 		// a record in the TVPairSet indicating that it should exist
 		tasksToAdd := []string{}
+		madeChanges := false
 		for _, taskname := range pairs.ExecTasks.TaskNames(b.BuildVariant) {
-			if _, ok := existingTasksIndex[taskname]; ok {
+			if t, ok := existingTasksIndex[taskname]; ok {
+				if !t.Activated && activated {
+					madeChanges = true
+					// update task activation for dependencies that already exist
+					if err = SetActiveState(&t, evergreen.User, true); err != nil {
```

Comment:
I don't totally follow the logic. How does this ensure that we allow activate dependencies and not every task?

Reply:
My understanding was that the only tasks in `pairs` will be the new tasks and the tasks added by [IncludeDependencies](https://github.com/evergreen-ci/evergreen/blob/1c87d4fff7405b0d35b8ad416a5cc7a12b70ad0b/model/generate.go#L219)

Reply:
Right, we only iterate through the relevant pairs.